api/repositories.py

```diff
@@ -29,6 +29,10 @@ class UserRepository(Repository):
     def __init__(self, mongo_collection, repository_collection):
         Repository.__init__(self, mongo_collection, repository_collection)
 
+    def update_user_account(self, name,password_hash,house_location,house_name):
+        self.collection.update_one({'name': name, 'password_hash': password_hash,
```

Comment:
Only update fields that are not 'None'. You can do that by constructing the fields of this dictionary one by one (only insert the fields that have a not-None value).